### PR TITLE
Applying patch from jack-h2ml to current wp version

### PR DESCRIPTION
**Pull request by gturpin-dev**
Trac ticket: https://core.trac.wordpress.org/ticket/57566

### CODE REVIEW

File: src/wp-includes/blocks.php [hunk 1]
```diff
@@ -134,17 +134,31 @@ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
...
+	static $stylesheet_path_norm = '';
+	if ( ! $template_path_norm || ! $stylesheet_path_norm ) {
+		$template_path_norm   = wp_normalize_path( get_template_directory() );
+		$stylesheet_path_norm = wp_normalize_path( get_stylesheet_directory() );
+	}
 	$script_path_norm = wp_normalize_path( realpath( dirname( $metadata['file'] ) . '/' . $script_path ) );
 
-	$is_core_block  = isset( $metadata['file'] ) && str_starts_with( $metadata['file'], $wpinc_path_norm );
-	$is_theme_block = str_starts_with( $script_path_norm, $theme_path_norm );
+	$is_core_block = isset( $metadata['file'] ) && 0 === strpos( $metadata['file'], $wpinc_path_norm );
```
Comment (mukeshpanchal27) [inline src/wp-includes/blocks.php]:
```suggestion
	$is_core_block = isset( $metadata['file'] ) && str_starts_with( $metadata['file'], $wpinc_path_norm );
```

In [58012](https://core.trac.wordpress.org/ticket/58012), core use `str_starts_with`

File: src/wp-includes/blocks.php [hunk 2]
```diff
@@ -134,17 +134,31 @@ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
...
+
+	// Determine if the block script was registered in a theme, by checking if the script path starts with either
+	// the parent (template) or child (stylesheet) directory path.
```
Comment (mukeshpanchal27) [inline src/wp-includes/blocks.php]:
```suggestion
	/*
	 * Determine if the block script was registered in a theme, by checking if the script path starts with either
	 * the parent (template) or child (stylesheet) directory path.
	 */
```

Use multiline comment

**Review by mukeshpanchal27 — COMMENTED** [review]
Thanks @gturpin-dev for the PR, left some initial feedback.

File: src/wp-includes/blocks.php (hunk 1 above)
Comment (gturpin-dev) [reply src/wp-includes/blocks.php]: You're right

**Review by tellthemachines — APPROVED** [review]
Code LGTM and can confirm it works as expected in testing. Thanks for fixing this!

**Comment by audrasjb** [comment]
Committed in https://core.trac.wordpress.org/changeset/56183

**Pull request closed without merging**